File: android2_lesson03/src/java/java/com/example/ama/android2_lesson03/utils/PermissionManager.java
package com.example.ama.android2_lesson03.utils;

import android.Manifest;
import android.app.Activity;
import android.content.pm.PackageManager;
import android.support.v4.content.ContextCompat;

import com.example.ama.android2_lesson03.PocketMap;
import com.tbruyelle.rxpermissions2.RxPermissions;

import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;

public class PermissionManager {

    public static final String FINE_LOCATION = Manifest.permission.ACCESS_FINE_LOCATION;

    public static Disposable requestPermission(Activity activity, String permission, final OnGrantResult callback) {
        RxPermissions rxPermissions = new RxPermissions(activity);
        return rxPermissions
                .request(permission)
                .subscribe(new Consumer<Boolean>() {
                    @Override
                    public void accept(Boolean aBoolean) throws Exception {
                        callback.sendResult(aBoolean);
                    }
                });
    }

    public static boolean checkLocationPermission() {
        return ContextCompat.checkSelfPermission(PocketMap.getInstance(), FINE_LOCATION)
                == PackageManager.PERMISSION_GRANTED;
    }

    public interface OnGrantResult {
        void sendResult(boolean granted);
    }
}

File: android2_lesson03/src/kotlin/java/com/example/ama/android2_lesson03/utils/PermissionManager.kt
package com.example.ama.android2_lesson03.utils

import android.app.Activity
import android.content.pm.PackageManager
import android.support.v4.content.ContextCompat
import com.example.ama.android2_lesson03.PocketMap
import com.tbruyelle.rxpermissions2.RxPermissions
import io.reactivex.disposables.Disposable

const val FINE_LOCATION = android.Manifest.permission.ACCESS_FINE_LOCATION

object PermissionManager {

    fun requestPermission(activity: Activity, permission: String, callback: (granted: Boolean) -> Unit): Disposable =
            RxPermissions(activity)
                    .request(permission)
                    .subscribe { it -> callback.invoke(it) }

    fun checkLocationPermission() =
            ContextCompat.checkSelfPermission(PocketMap.instance, FINE_LOCATION) ==
                    PackageManager.PERMISSION_GRANTED
}